OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -1085,6 +1085,36 @@ internal static void UpgradeActorRules(int engineVersion, ref List<MiniYamlNode>
 					}
 				}
 
+				if (engineVersion < 20150607)
+				{
+					// Add WithRankDecoration to all actors using GainsExperience
+					var ge = node.Value.Nodes.FirstOrDefault(n => n.Key == "GainsExperience");
+					if (ge != null)
+					{
+						var nodeUpgrades = ge.Value.Nodes.FirstOrDefault(n => n.Key == "Upgrades");
+						var upgrades = nodeUpgrades != null ? nodeUpgrades.Value.Nodes.Count() : 4;
+
+						if (upgrades != 0)
+						{
+							var nodeChPal = ge.Value.Nodes.FirstOrDefault(n => n.Key == "ChevronPalette");
+							var chPal = nodeChPal != null && !string.IsNullOrEmpty(nodeChPal.Value.Value) ? nodeChPal.Value.Value : "effect";
+
+							node.Value.Nodes.Add(new MiniYamlNode("WithRankDecoration", null, new List<MiniYamlNode>
+							{
+								new MiniYamlNode("Image", "rank"),
+								new MiniYamlNode("Sequence", "rank"),
+								new MiniYamlNode("Palette", chPal),
+								new MiniYamlNode("ReferencePoint", "BottomRight"),
```

Comment:
Are you sure? The `Rank` effect has drawn the chevron in the bottom right corner...